core/meta/inc/TInterpreter.h

```diff
@@ -510,4 +513,25 @@ R__EXTERN TInterpreter* (*gPtr2Interpreter)();
 R__EXTERN TInterpreter* gCling;
 #endif
 
+namespace ROOT {
+namespace Internal {
+struct DeregisterInterpreterMutexRAII {
```

Comment:
This sounds like the "interpreter mutex" is deregistered (aka removed) from some list/white-board/something ... This does not sound like what it actually does.  Is there a better name and/or am I missing something?